### PR TITLE
Field Can Be Readonly

### LINKED ISSUE #223: Implement FieldCanBeMadeReadonly
A field that is only written to inside constructors or inline, should be made `readonly`.

### PR DESCRIPTION
**Pull request by Hosch250**
Close #223

### CODE REVIEW

**Comment by Hosch250** [comment]
Todo: static fields assigned from non-static ctor cannot be readonly.

**Comment by Hosch250** [comment]
Todo: Static readonly field cannot be assigned from non-static ctor.

File: VSDiagnostics/VSDiagnostics/VSDiagnostics.Test/Tests/General/FieldCanBeReadonlyTests.cs [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+    class MyClass
+    {
+        private readonly int _foo = 0, _bar = 1;
+    }
+}";
+
+            VerifyDiagnostic(original,
+                string.Format(FieldCanBeReadonlyAnalyzer.Rule.MessageFormat.ToString(), "_foo"),
+                string.Format(FieldCanBeReadonlyAnalyzer.Rule.MessageFormat.ToString(), "_bar"));
+            VerifyFix(original, result);
```
Comment (JohanLarsson) [inline VSDiagnostics/VSDiagnostics/VSDiagnostics.Test/Tests/General/FieldCanBeReadonlyTests.cs]: Why not `VerifyFix(original, original);`? Clearer & shorter IMO.
Or maybe a `VerifyNotFixed(original);`?

**Comment by Hosch250** [comment]
This has an issue with crashing on nodes in generated files in WPF (essentially the same problem as #614).  It can't find the node for some reason or other, but it doesn't matter because we don't need to analyze it.

**Pull request closed without merging**